src/fu-engine.c

```diff
@@ -2696,6 +2715,9 @@ fu_engine_activate (FuEngine *self, const gchar *device_id, GError **error)
 	if (!fu_plugin_runner_activate (plugin, device, error))
 		return FALSE;
 
+	/* unconditionally do this on success */
+	fu_device_remove_flag (device, FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION);
```

Comment:
I don't think this is correct, this is already done in `fu_plugin_runner_activate`

Reply:
Agree, dropped, thanks.